bin/cleanupSamplesheets.sh

```diff
@@ -0,0 +1,215 @@
+	l4b_log_level="DEBUG"
+	l4b_log_level_prio=${l4b_log_levels[${l4b_log_level}]}
+
+else
+	dryrun="no"
+fi
+
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "dryrun=${dryrun}"
+
+#
+# Source config files.
+#
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Sourcing config files ..."
+declare -a configFiles=(
+	"${CFG_DIR}/${group}.cfg"
+	"${CFG_DIR}/${HOSTNAME_SHORT}.cfg"
+	"${CFG_DIR}/sharedConfig.cfg"
+)
+for configFile in "${configFiles[@]}"; do
+	if [[ -f "${configFile}" && -r "${configFile}" ]]; then
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Sourcing config file ${configFile} ..."
+		#
+		# In some Bash versions the source command does not work properly with process substitution.
+		# Therefore we source a first time with process substitution for proper error handling
+		# and a second time without just to make sure we can use the content from the sourced files.
+		#
+		# Disable shellcheck code syntax checking for config files.
+		# shellcheck source=/dev/null
+		mixed_stdouterr=$(source "${configFile}" 2>&1) || log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" "${?}" "Cannot source ${configFile}."
+		# shellcheck source=/dev/null
+		source "${configFile}"  # May seem redundant, but is a mandatory workaround for some Bash versions.
+	else
+		log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "Config file ${configFile} missing or not accessible."
+	fi
+done
+
+
+pipeline=$(echo "${REPLACEDPIPELINECOLUMN}" | cut -d'+' -f1)
+# shellcheck disable=SC2029
+mapfile -t rawdataSamplesheets < <(ssh "${DATA_MANAGER}"@"${HOSTNAME_TMP}" "find \"${TMP_ROOT_DIAGNOSTICS_DIR}/Samplesheets/${pipeline}/\" -maxdepth 1 -mindepth 1 -type f")
+if [[ "${#rawdataSamplesheets[@]}" -eq '0' ]]
+then
+	log4Bash 'WARN' "${LINENO}" "${FUNCNAME:-main}" '0' "No rawdataSamplesheets files found @ ${TMP_ROOT_DIAGNOSTICS_DIR}/Samplesheets/${pipeline}."
+	exit
+else
+	for rawdataSamplesheet in "${rawdataSamplesheets[@]}"
+	do
+		rawdata=$(basename "${rawdataSamplesheet}" .csv)
+		log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Found ${rawdata} on ${TMP_ROOT_DIAGNOSTICS_DIR}, check if it is present on ${PRM_ROOT_DIR}"
+
+		for prm_dir in "${ALL_PRM[@]}"
+		do
+			log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "looping through ${prm_dir}"
+			export PRM_ROOT_DIR="/groups/${group}/${prm_dir}/"
+			if [[ "${pipeline}" == "NGS_Demultiplexing" ]]
+			then
+				if [[ -e "${PRM_ROOT_DIR}/rawdata/${PRMRAWDATA}/${rawdata}/" ]]
+				then
+					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "Great, the rawdata of ${rawdata} is already processed and stored on ${PRM_ROOT_DIR}"
+					if [[ "${dryrun}" == "no" ]]
+					then
+						log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "time to remove the extra samplesheets from ${TMP_ROOT_DIAGNOSTICS_DIR}/Samplesheets/${pipeline}/"
+						# shellcheck disable=SC2029
+						ssh "${DATA_MANAGER}"@"${HOSTNAME_TMP}" "rm \"${TMP_ROOT_DIAGNOSTICS_DIR}/Samplesheets/${pipeline}/${rawdata}\".csv"
```

Comment:
hier zou ik in dit geval wel `rm -f` gebruiken, als jij namelijk niet de eigenaar van de file bent kan ie een prompt geven die je niet wilt.